client/src/_shared/_hooks/useAuthentication.js

```diff
@@ -0,0 +1,24 @@
+import dayjs from 'dayjs'
+
+export function useAuthentication() {
+  const setToken = (token, expiration = dayjs().add('1', 'day')) => {
+    localStorage.setItem('pie-token', token)
+    localStorage.setItem('pie-expiration', expiration)
```

Comment:
**question**: Can't we decode the JWT client-side (we don't need the secret key because we're not validating the token) using [jwt-decode](https://github.com/auth0/jwt-decode)  and extract expiration `exp` from the token?

Reply:
Oooooooooh interesting point.  I guess I hadn't thought of it that way.  I'm going to investigate!  Thank you!